### DISCUSSION TITLE
The route duplicating it self

### DISCUSSION THREAD

**Discussion by sukmaw** [inertiajs/inertia#1145] [Help (React)]
### Laravel version

v8.75

### inertia version
"@inertiajs/inertia": "^0.10.0",
"@inertiajs/inertia-react": "^0.8.0",
"@inertiajs/progress": "^0.2.6",

First the url look right like this
![Screenshot from 2022-03-15 02-08-00](https://user-images.githubusercontent.com/12524835/158244082-91132b21-c851-47c5-aa47-b1064e9f0b0c.png)

But after several second it change and duplicating the url, like this
![Screenshot from 2022-03-15 02-06-16](https://user-images.githubusercontent.com/12524835/158244206-1d995dee-c88e-478b-a074-0c9952ed77b9.png)

im trying to trace it
first, i find the `window.history.pushState` function on app js and unminifying the scripts, and i try debugging it (actual just console log the param)
i found this `public/app.js`
![Screenshot from 2022-03-15 17-54-34](https://user-images.githubusercontent.com/12524835/158362771-db99c1d2-20c9-45f2-a28d-f529c805809c.png)

and here the result of console 

![Screenshot from 2022-03-15 17-55-17](https://user-images.githubusercontent.com/12524835/158362870-bcbb8902-2c6a-431c-9948-90f416f77d5d.png)

and the first comment of that code is the inertiajs

![Screenshot from 2022-03-15 17-56-22](https://user-images.githubusercontent.com/12524835/158363060-366ccfc9-8774-497f-9082-5293a53b6da7.png)

if i see on the `resource/app.js` i think on the createInertiaApp
![Screenshot from 2022-03-15 17-58-52](https://user-images.githubusercontent.com/12524835/158363442-1295a3ad-2405-4f6e-8bef-adca5abeeebe.png)

### NOTE : 
But if i run it on the root web server or developing with php web server it works
But my architecture is put it on the subfolder

**Comment by Neeraj1005** [comment]
@sukmaw did you resolved this issue...?
because I'm facing the same issue in my app.

**Reply by sukmaw** [reply]
nope

**Reply by sukmaw** [reply]
i think we must re-architecturing your project
like use subdomain instead subfolder

**Reply by shantanu-taskmasterpro** [reply]
Add @php($page['url'] = App::make('request')->getRequestUri()) this code at the start of your app.blade.php (where you are loading the app.js)

**Reply by Neeraj1005** [reply]
@shantanu-taskmasterpro can you share some screenshot of code?? in you app.js

**Comment by ReactJsScc** [comment]
any one solved this issue ?

**Reply by shantanu-taskmasterpro** [reply]
Go to your vendor folder (  \vendor\inertiajs\inertia-laravel\src\Response.php) In Response.php change toResponse method to this

 public function toResponse($request)
    {
        $only = array_filter(explode(',', $request->header('X-Inertia-Partial-Data', '')));

        $props = ($only && $request->header('X-Inertia-Partial-Component') === $this->component)
            ? Arr::only($this->props, $only)
            : array_filter($this->props, static function ($prop) {
                return ! ($prop instanceof LazyProp);
            });

        $props = $this->resolvePropertyInstances($props, $request);

        $page = [
            'component' => $this->component,
            'props' => $props,
            'url' => $request->getRequestUri(), // originally it is like $request->getBaseUrl().$request->getRequestUri(),
            'version' => $this->version,
        ];

        if ($request->header('X-Inertia')) {
            return new JsonResponse($page, 200, ['X-Inertia' => 'true']);
        }

        return ResponseFactory::view($this->rootView, $this->viewData + ['page' => $page]);
    }
    
    If you do not want to change the core file, then create Response.php under laravel/app/(name of directory)/Response.php
    and paste all the content from original Response.php with above changes and add following attributes under autoload object 
    inside composer.json
   
   
    "autoload": {
      "files": [
             
            "app/<directory-name>/Response.php"
        ],
        "exclude-from-classmap": [
            "laravel/vendor/inertiajs/inertia-laravel/src/Response.php"
        ]
      }
    
    Hope this will help.
